src/twisted/internet/tcp.py

```diff
@@ -853,6 +858,246 @@ def getPeer(self):
 
 
 
+def _silenceOutputs(outputs):
+    """
+    An L{automat} output collector that returns L{None}.
+
+    @param outputs: An outputs iterator
+
+    @return: L{None}
+    """
+    list(outputs)
```

Comment:
I read through the automat docs (all of them, I think) to try to understand this line but I didn't find anything that explains why this is a useful thing to do.  Did I miss something from the docs?

If I didn't miss anything, please file a doc issue against automat to explain the issue this code is addressing and also add a comment here meanwhile.

Reply:
https://github.com/glyph/automat/issues/18 is half the issue here and all of the issue with `nthOutput`.  This collector also ensures that the transition it's applied to returns `None`; by default, Automat inputs return a (possibly empty) list of their output method's return values, which makes no sense for, say, '_FileDescriptorReservation.acquire`.